prog/postgres/postgres_server_nexus.rb

```diff
@@ -89,32 +91,72 @@ def before_run
   label def install_postgres
     case vm.sshable.cmd("common/bin/daemonizer --check install_postgres")
     when "Succeeded"
-      hop_refresh_certificates
+      hop_install_walg
     when "Failed", "NotStarted"
       vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/install_postgres' install_postgres")
     end
 
     nap 5
   end
 
+  label def install_walg
+    case vm.sshable.cmd("common/bin/daemonizer --check install_wal-g")
+    when "Succeeded"
+      refresh_walg_credentials
+
+      hop_initialize_empty_database if postgres_server.primary?
+      hop_initialize_database_from_backup
+    when "Failed", "NotStarted"
+      vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/install-wal-g c56a2315d3a63560f0227cb0bf902da8445963c7' install_wal-g")
+    end
+
+    nap 5
+  end
+
+  label def initialize_empty_database
+    case vm.sshable.cmd("common/bin/daemonizer --check initialize_empty_database")
+    when "Succeeded"
+      hop_refresh_certificates
+    when "Failed", "NotStarted"
+      vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/initialize-empty-database' initialize_empty_database")
+    end
+
+    nap 5
+  end
+
+  label def initialize_database_from_backup
+    case vm.sshable.cmd("common/bin/daemonizer --check initialize_database_from_backup")
+    when "Succeeded"
+      hop_refresh_certificates
+    when "Failed", "NotStarted"
+      backup_label = postgres_server.timeline.last_backup_label_before_target(target: postgres_server.resource.restore_target)
+      vm.sshable.cmd("common/bin/daemonizer 'sudo postgres/bin/initialize-database-from-backup #{backup_label}' initialize_database_from_backup")
+    end
+
+    nap 5
+  end
+
```

Comment:
dumb thought, these could be parallel, but perhaps not yet, in general we're being excessively linear and it's slowing everything down, including iteration. But, we haven't yet made our parallelism model super mature (host nexus tries a bit with `learn` from ancient days, `dnsmasq` tries a different approach to get parallelism in the same prog, we never worked on making exitval/reap more ergonomic)

But, the code base is bigger, it's becoming, perhaps, more profitable to figure this out, if for no other reason that speed of iteration, though customers will like it too. 

Just a thought. I'm linking this to my journal.